src/App.jsx

```diff
@@ -15,12 +15,37 @@ function App() {
   function handleDeleteTodo(id) {
     setTodos(todos.filter((todo) => todo.id !== id));
   }
+  function handleToggleTodo(id) {
+    setTodos(
+      todos.map((todo) =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo
+      )
+    );
+  }
+  function handleEdit(id, newContent) {
+    setTodos(
+      todos.map((todo) =>
+        id === todo.id ? { ...todo, content: newContent } : todo
+      )
+    );
+  }
```

Comment:
These two functions are quite similar: they locate the target todo, modify it, and return a new array. We can simplify this with an abstraction:

```js
const updateTodoList = (targetId, updateFn) => {
    setTodos(
      todos.map((todo) =>
        targetId === todo.id ? updateFn(todo) : todo
      )
    );
}

const handleEditTodo = (id, newContent) => {
  updateTodoList(id, todo => ({ ...todo, content: newContent }))
};

const handleToggleTodo = (id) => {
  updateTodoList(id, todo => ({ ...todo, completed: !todo.completed }))
}
```